controllers/openstackdataplanenodeset_controller.go

```diff
@@ -417,6 +417,24 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, nil
 	}
 
+	isDeploymentReady, isDeploymentRunning, isDeploymentFailed, err := checkDeployment(helper, instance)
+	if err != nil {
+		instance.Status.Conditions.MarkFalse(
+			condition.DeploymentReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			condition.DeploymentReadyErrorMessage,
+			err.Error())
+		Log.Error(err, "Unable to get deployed OpenStackDataPlaneDeployments.")
+		return ctrl.Result{}, err
+	}
+
+	if isDeploymentRunning {
+		Log.Info("Deployment running, requeueing...", "instance", instance)
+		return ctrl.Result{RequeueAfter: time.Second * 15}, nil
```

Comment:
Though this probably won't be required after the restriction in the webhook, we can still have this if we want. However, re-queuing after 15 seconds, when a deployment is running is probably not ideal.

Reply:
In deployment we have been using `DeploymentRequeueTime`[0] which is set by default to 15 sec. That already takes care of cases when nodeset doesn't exist or when it doesn't have the right conditions, but that is largely orthogonal to this issue.

I would suggest that this delay is set to some other time. Perhaps turning it into parameter so we can adjust it later, if it turns out our estimates were off.


[0] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/openstackdataplanedeployment_types.go#L57